src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -1059,56 +1057,51 @@ private void runNextPiece(
         logger.debug("Init progress: {}, taskState:{}, task id: {}", initProgress, taskState, taskId);
 
         if (pieceStartTime < dataEndTime) {
```

Comment:
should pieceStartTime be pieceEndTime?

Reply:
Explain in an example:

`pieceStartTime` : 2021-07-01 10:00:00
`pieceEndTime` : 2021-07-02 10:00:00
`dataEndTime` : 2021-07-02 01:00:00

`pieceEndTime` > `dataEndTime` , but `pieceStartTime` < `dataEndTime`, we should detect the data between `pieceStartTime` and `dataEndTime`

Reply:
You passed in pieceEndTime to the argument pieceStartTime in the runNextPiece function.  Why are they inconsistent?

```
anomalyResultBulkIndexHandler
            .bulkIndexAnomalyResult(
                anomalyResults,
                new ThreadedActionListener<>(logger, threadPool, AD_BATCH_TASK_THREAD_POOL_NAME, ActionListener.wrap(r -> {
                    try {
                        runNextPiece(adTask, pieceEndTime, dataStartTime, dataEndTime, interval, internalListener);
                    } catch (Exception e) {
                        internalListener.onFailure(e);
                    }
                }, e -> {
                    logger.error("Fail to bulk index anomaly result", e);
                    internalListener.onFailure(e);
                }), false)
            );
```

Reply:
Current piece end time is the next piece's start time

Reply:
ah, could you add comments to explain?

Reply:
ok